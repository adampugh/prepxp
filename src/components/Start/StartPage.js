import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";
 
import * as actions from "../../store/actions/actions";
import Navbar from "../UI/navbarLoggedIn";
import StartGrid from "./startGrid";
import StartComplete from "./startComplete";
import StartSavedModal from "./startSavedModal";
import Footer from "../UI/footer";

export class StartPage extends Component {
    state = {
        questionsComplete: false,
        modalIsOpen: false
    }

    componentDidMount() {
        window.scrollTo(0, 0);
    }

    finishQuestions = () => {
        window.scrollTo(0, 0);
        this.setState({
            questionsComplete: true
        })
    }

    openModal = () => {
        this.setState({
            modalIsOpen: true
        });
    }

    closeModal = () => {
        this.setState({
            modalIsOpen: false
        })
    }

    render(props) {
        // let content =  this.props.list === null ?  : startPage;
        // let startPage = (
            
        // );
        // let redirect = <Redirect to="/dashboard" />;
        
        return (
            <div>
                {
                    this.props.list === null 
                        ? <Redirect to="/dashboard" />
                        : <div>
                            <StartSavedModal 
                                isOpen={this.state.modalIsOpen}
                                closeModal={this.closeModal} />
                            <Navbar />
                            {this.state.questionsComplete ? 
                                <StartComplete /> :
                                <StartGrid 
                                    finishQuestions={this.finishQuestions}
                                    questions={this.props.list.questions}
                                    id={this.props.list.id}
                                    startSaveAnswer={this.props.startSaveAnswer}
                                    closeModal={this.closeModal}
                                    openModal={this.openModal}
                                />
                            }
                            <Footer />
                        </div>
                }
            </div>
        )
    } 
}


const mapStateToProps = (state, ownProps) => {
    return {
        list: ownProps.location.state ? state.lists.filter(list => list.id === ownProps.location.state.list.id)[0] : null
    }
}

const mapDispatchToProps = dispatch => {
    return {
        startSaveAnswer: (id, index, answer, questionId) => dispatch(actions.startSaveAnswer(id, index, answer, questionId))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(StartPage);