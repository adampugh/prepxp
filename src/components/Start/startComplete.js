import React from "react";
import { Link } from "react-router-dom";
 
import interview from "../../assets/images/interview2.png"

const startComplete = () => (
    <section className="startComplete__div">
        <div className="container">
            <div className="startComplete__grid">
                <div className="startComplete__grid__text">
                    <h1>Questions complete</h1>
                    <p> Well done you have completed this list, good luck in your upcoming interviews!</p>
                    <div className="startComplete__buttons">
                        <Link to="/dashboard">
                            <button className="btn">Dashboard</button>
                        </Link>
                    </div>
                </div>
                <img src={interview} alt="people conducting an interview" />
            </div>
        </div>
    </section>
);

export default startComplete;