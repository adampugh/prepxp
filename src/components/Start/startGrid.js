import React, { Component } from "react";

class startGrid extends Component {
    state = {
        currentIndex: 0,
        showAnswer: false,
        answer: ""
    }

    updateAnswer = (e) => {
        if (e.target.value.length > 0) {
            this.setState({
                answer: e.target.value
            })
        }
    }

    showAnswer = () => {
        this.setState({
            showAnswer: !this.state.showAnswer
        })
    }

    nextQuestion = () => {
        if (this.state.currentIndex + 1 === this.props.questions.length) {
            this.props.finishQuestions();
        } else {
            document.querySelector("textarea").value = "";
            // window.scrollTo(0, 0);
            this.setState({
                currentIndex: this.state.currentIndex + 1,
                showAnswer: false
            });
        }
    }

    handleStartSaveAnswer = () => {
        if (this.state.answer.length > 0) {
            this.props.startSaveAnswer(
                this.props.id, 
                this.state.currentIndex, 
                this.state.answer, 
                this.props.questions[this.state.currentIndex].id
            );
        }
        document.getElementById("answer__textarea").value = "";
        this.setState({answer: ""});
        this.showSavedModal();
    }

    showSavedModal = () => {
        this.props.openModal();
        setTimeout(() => {
            this.props.closeModal();
            this.nextQuestion();
        }, 1000);
    }

    render() {
        return (
            <section className="start">
                <div className="container-s">
                    <div className="start__box">
                        <div className="start__questionBox">
                            <h2>
                                <span>{`Q${this.state.currentIndex + 1} `}</span> 
                                {this.props.questions[this.state.currentIndex].question}
                            </h2>
                            <button className="btn" onClick={this.showAnswer}>
                                {this.state.showAnswer ? "Hide answer" : "Show answer"}
                            </button>
                        </div>
                        <div className={this.state.showAnswer 
                                ? "start__previousAnswer start__previousAnswer--purple"
                                : "start__previousAnswer start__previousAnswer--black" }>
                            {
                                this.state.showAnswer 
                                    ? this.props.questions[this.state.currentIndex].answer === "" 
                                    || this.props.questions[this.state.currentIndex].answer === undefined
                                        ? <p>You haven't added an answer yet</p>
                                        : <p>{this.props.questions[this.state.currentIndex].answer}</p> 
                                    : <p>Answer hidden</p>
                            }
                        </div>
                        <div className="start__answerBox">
                            <textarea 
                                id="answer__textarea"
                                onChange={(e) => this.updateAnswer(e)}/>
                        </div>
                        <div className="start__button__div">
                            <button 
                                className="btn"
                                onClick={this.handleStartSaveAnswer} >Save</button>
                            <button 
                                className="btn" 
                                onClick={this.nextQuestion}>Next</button>
                        </div>
                    </div>
                </div>
            </section>
        )
    }
}

export default startGrid;