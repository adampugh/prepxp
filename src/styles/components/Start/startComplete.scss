.startComplete__div {
    background: $purple-gradient, url("../assets/images/bg5.jpg") no-repeat center;
    background-size: cover;
    color: white;
    padding: $padding-m 0;
    h1 {
        @include header;
    }
    p {
        font-size: $font-s;
    }
    img {
        width: 100%;
    }
}

.startComplete__grid {
    display: grid;
    grid-template-columns: 1fr;
}

.startComplete__grid__text {
    margin-top: $padding-m;
}

.startComplete__buttons {
    margin: $padding-s 0 $padding-m 0;
    button {
        display: block;
    }
}

@media (min-width: $desktop-breakpoint) {
    .startComplete__div {
        padding-top: 120px;
    }

    .startComplete__grid {
        grid-template-columns: 1fr 1fr;
    }

    .startComplete__buttons {
        button {
            margin-right: 10px;
        }
    }
}