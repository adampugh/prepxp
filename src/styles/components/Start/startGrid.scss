.start {
    background: $purple-gradient, url("../assets/images/bg5.jpg") no-repeat center;
    background-size: cover;
    padding: $padding-m 0;
    padding-top: 110px;
}

.start__questionBox {
    border-radius: 20px;
    word-break: break-word;
    span {
        color: $off-purple;
    }
}

.start__previousAnswer {
    color: white;
    padding: $padding-s;
    border-radius: 20px;
    margin: $padding-s 0;
    font-size: $font-s;
    word-break: break-word;
    p {
        margin: 0;
    }
}

.start__previousAnswer--black {
    background: $black-gradient, url("../assets/images/bg5.jpg") no-repeat center center;
}
.start__previousAnswer--purple {
    background: $purple-gradient, url("../assets/images/bg5.jpg") no-repeat center;
}

.start__answerBox {
    margin: 0 auto;
    text-align: center;
    textarea {
        width: 100%;
        min-height: 150px;
        border: 2px solid $light-grey;
        border-radius: 20px;
        padding: 20px;
        box-sizing: border-box;
    }
}

.start__button__div {
    display: flex;
    justify-content: space-between;
    margin: $padding-s 0;
}

.start__box {
    padding: $padding-s;
    background: white;
    border-radius: 20px;
    @include boxshadow;
}

@media (min-width: $desktop-breakpoint) {
    .start__questionBox {
        padding: $padding-s 0;
    }

    .start__previousAnswer {
        padding: $padding-s;
        margin: $padding-s 0;
    }

    .start__box {
        padding: 10px $padding-m;
    }
}