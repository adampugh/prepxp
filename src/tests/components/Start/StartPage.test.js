import React from "react";
import { shallow } from "enzyme";

import { StartPage } from "../../../components/Start/StartPage";
import StartComplete from "../../../components/Start/startComplete";
import user from "../../fixtures/userLists";

describe("<StartPage />", () => {
    const locationObj = {
        state: {
            list: user.lists[0]
        }
    }

    it("should match snapshot", () => {
        const wrapper = shallow(<StartPage location={locationObj} list={user.lists[0]}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render startComplete if questions are complete", () => {
        const wrapper = shallow(<StartPage location={locationObj} list={user.lists[0]}/>);
        wrapper.setState({questionsComplete: true});
        expect(wrapper.contains(<StartComplete />)).toBe(true);
    });

})