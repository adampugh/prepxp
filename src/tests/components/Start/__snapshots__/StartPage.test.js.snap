// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartPage /> should match snapshot 1`] = `
<div>
  <div>
    <startSavedModal
      closeModal={[Function]}
      isOpen={false}
    />
    <Connect(Navbar) />
    <startGrid
      closeModal={[Function]}
      finishQuestions={[Function]}
      id="234"
      openModal={[Function]}
      questions={
        Array [
          Object {
            "answer": "JavaScript is a language for the web.",
            "question": "What is the difference between let and var?",
          },
          Object {
            "answer": "Brendan Eich",
            "question": "Who created JavaScript?",
          },
        ]
      }
    />
    <footer />
  </div>
</div>
`;
